application/sandbox/serial/process/src/services/service.sessions.ts

```diff
@@ -17,6 +44,7 @@ class ServiceSessions {
         PluginIPCService.subscribe(IPCMessages.PluginInternalMessage, this._onIncomeRenderIPCMessage);
         PluginIPCService.on(PluginIPCService.Events.openStream, this._onOpenStream);
         PluginIPCService.on(PluginIPCService.Events.closeStream, this._onCloseStream);
+        ServiceConfig.setDefault<{[ports:string]: {[port: string]: IPortOptions}}>( { ports: {} } );
```

Comment:
I would recommend to create an interface for settings and add one more level, like
```
interface IRecentPortData {
    [ports: string]: {[port: string]: IPortOptions };
}

interface IPluginSettings {
    recent: IRecentPortData;
}
```

Also, not clear, what does mean `ports:string`. Is it some kind of hash? or?

Reply:
`{ "ports":                  // <-- [ports: string]
    { "COM4": { ... } }
    { "COM1": { ... } }
    { "COM5": { ... } }
}`

Reply:
Above is the content of the config file, so the `ports:string` is the key for it to read the ports

Reply:
`interface IPluginSettings {`
`    recent: IRecentPortData;`
`}`
I'm not sure where this interface would help

Reply:
Would it maybe not be better to do it like this:
`interface IRecentPortData {`
`    [port: string]: IOptions;`
`}`
`interface IPluginSettings {`
`    recent: IRecentPortData;`
`}`